dcmdump/src/main.rs

```diff
@@ -29,6 +30,44 @@ const ERROR_READ: i32 = -2;
 /// Exit code for when an error emerged while dumping the file.
 const ERROR_PRINT: i32 = -3;
 
+#[derive(Debug)]
+struct ColoringError { }
+
+#[derive(Clone, Copy, Debug)]
+enum Coloring {
+    Never,
+    Auto,
+    Always,
+}
+
+impl ToString for ColoringError {
+    fn to_string(&self) -> String {
+        "invalid color mode".into()
+    }
+}
```

Comment:
Error types should implement `Display`, which is also a requirement for implementing `Error`.
 
```suggestion
impl Display for ColoringError {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        f.write_str("invalid color mode")
    }
}

impl std::error::Error for Coloring {}
```